### ISSUE TITLE
Random values are not consistent across usages

### ISSUE THREAD

**Issue by Bendzae** [djeedai/bevy_hanabi#247]
**Crate versions**
`bevy` version: 0.11.3
`bevy_hanabi` version: 0.7

**Describe the bug**
Given an expression with a random value e.g. `let random_expression = w.lit(2.0) * w.rand(ScalarType::Float)`:
When using this expression multiple times e.g.
```rust
let particle_height = random_expression * w.lit(1.5);
let particle_width = random_expression * w.lit(2.5);
```
For each use the expression (`random_expression`) evaluates to a different value.
This happens, because the resulting shadercode replaces the the `rand(...)` with a new invocation of the corresponding random function.

**Expected behavior**
The expression including the random expression should evaluate to the same value for each usage (for a single particle). 

**To Reproduce**
See code in the description.

**Comment by djeedai** [comment]
Duplicate of #240, fixed in #241.

**Comment by Bendzae** [comment]
Ah I didn't realize you fixed it already. Thanks a lot!
